src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.cs

```diff
@@ -1187,14 +928,16 @@ private static int ComputeTimeoutPoint(int timeoutLength)
         /// <param name="localFinally">A cleanup function to destroy thread local state.</param>
         /// <remarks>Only one of the body arguments may be supplied (i.e. they are exclusive).</remarks>
         /// <returns>A <see cref="System.Threading.Tasks.ParallelLoopResult"/> structure.</returns>
-        private static ParallelLoopResult ForWorker64<TLocal>(
-            long fromInclusive, long toExclusive,
+        private static ParallelLoopResult ForWorker<TLocal, TInt>(
+            TInt fromInclusive, TInt toExclusive,
             ParallelOptions parallelOptions,
-            Action<long>? body,
-            Action<long, ParallelLoopState>? bodyWithState,
-            Func<long, ParallelLoopState, TLocal, TLocal>? bodyWithLocal,
+            Action<TInt>? body,
+            Action<TInt, ParallelLoopState>? bodyWithState,
+            Func<TInt, ParallelLoopState, TLocal, TLocal>? bodyWithLocal,
             Func<TLocal>? localInit, Action<TLocal>? localFinally)
+            where TInt : struct, INumber<TInt>, IMinMaxValue<TInt>
```

Comment:
Should this be:
```suggestion
            where TInt : struct, IBinaryInteger<TInt>, IMinMaxValue<TInt>
```
?

Reply:
Just because it's more specific, right? Sure, I can change it.